### ISSUE TITLE
'Invalid array dimensions'

### ISSUE THREAD

**Issue by jamii** [JuliaStats/Feather.jl#21]
``` julia
julia> using Feather

julia> import SQLite

julia> db = SQLite.DB("../imdb/imdb.sqlite")
SQLite.DB("../imdb/imdb.sqlite")

julia> @time Feather.write("../imdb/imdb.feather", SQLite.Source, db, "select * from movie_info")
 54.178787 seconds (369.61 M allocations: 11.243 GB, 15.79% gc time)
Feather.Sink(Data.Schema:
rows: 14835720	cols: 5
Columns:
 "id"            Nullable{Int64} 
 "movie_id"      Nullable{Int64} 
 "info_type_id"  Nullable{Int64} 
 "info"          Nullable{String}
 "note"          Nullable{String},Feather.Metadata.CTable("",14835720,Feather.Metadata.Column[Feather.Metadata.Column("id",Feather.Metadata.PrimitiveArray(INT64,PLAIN,8,14835720,0,118685760),0,nothing,""),Feather.Metadata.Column("movie_id",Feather.Metadata.PrimitiveArray(INT64,PLAIN,118685768,14835720,0,118685760),0,nothing,""),Feather.Metadata.Column("info_type_id",Feather.Metadata.PrimitiveArray(INT64,PLAIN,118685768,14835720,0,118685760),0,nothing,""),Feather.Metadata.Column("info",Feather.Metadata.PrimitiveArray(UTF8,PLAIN,118685768,14835720,0,685698016),0,nothing,""),Feather.Metadata.Column("note",Feather.Metadata.PrimitiveArray(UTF8,PLAIN,1042080099,14835720,0,86452160),0,nothing,"")],2,""),IOStream(<file ../imdb/imdb.feather>),"","")

julia> @time Feather.read("../imdb/imdb.feather")
ERROR: invalid Array dimensions
 in (::Feather.##15#16{Array{Int32,1},Array{UInt8,1}})(::Int64) at ./<missing>:0
 in collect(::Base.Generator{UnitRange{Int64},Feather.##15#16{Array{Int32,1},Array{UInt8,1}}}) at ./array.jl:307
 in getcolumn(::Feather.Source, ::Type{WeakRefString{UInt8}}, ::Int64) at /home/jamie/.julia/v0.5/Feather/src/Feather.jl:153
 in stream!(::Feather.Source, ::Type{DataStreams.Data.Column}, ::DataFrames.DataFrame, ::Bool) at /home/jamie/.julia/v0.5/DataStreams/src/DataStreams.jl:292
 in stream!(::Feather.Source, ::Type{DataFrames.DataFrame}, ::Bool) at /home/jamie/.julia/v0.5/DataStreams/src/DataStreams.jl:112
 in read(::String) at /home/jamie/.julia/v0.5/Feather/src/Feather.jl:217
```

Where movie_info was loaded from movie_info.csv in http://homepages.cwi.nl/%7Eboncz/job/imdb.tgz

``` julia
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753 (2016-09-19 18:14 UTC)
Platform Info:
  System: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E3-1505M v5 @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, broadwell)

julia> Pkg.status()
13 required packages:
 - Atom                          0.5.2
 - Benchmark                     0.1.0
 - BenchmarkTools                0.0.5
 - Blink                         0.4.1
 - Feather                       0.1.5
 - Gadfly                        0.4.4
 - Gallium                       0.0.4
 - Hiccup                        0.0.3
 - JLD                           0.6.4
 - Match                         0.3.0
 - ProfileView                   0.1.4
 - PyPlot                        2.2.4
 - SQLite                        0.3.7
85 additional packages:
 - ASTInterpreter                0.0.4
 - AbstractTrees                 0.0.4
 - ArgParse                      0.3.1
 - AxisAlgorithms                0.1.5
 - BinDeps                       0.4.5
 - Blosc                         0.1.7
 - COFF                          0.0.2
 - CRC                           1.1.1
 - Cairo                         0.2.35
 - Calculus                      0.1.15
 - CategoricalArrays             0.0.4
 - CodeTools                     0.4.1
 - Codecs                        0.2.0
 - ColorTypes                    0.2.8
 - Colors                        0.6.8
 - Compat                        0.9.2
 - Compose                       0.4.4
 - Conda                         0.3.2
 - Contour                       0.2.0
 - DWARF                         0.1.0
 - DataArrays                    0.3.8
 - DataFrames                    0.8.3
 - DataStreams                   0.0.15
 - DataStructures                0.4.6
 - Distances                     0.3.2
 - Distributions                 0.10.2
 - Docile                        0.5.23
 - ELF                           0.1.0
 - FileIO                        0.1.2
 - FixedPointNumbers             0.1.8
 - FixedSizeArrays               0.2.3
 - FlatBuffers                   0.1.2
 - ForwardDiff                   0.2.5
 - GZip                          0.2.20
 - Graphics                      0.1.3
 - Gtk                           0.10.2
 - GtkUtilities                  0.1.0
 - HDF5                          0.6.6
 - Hexagons                      0.0.4
 - HttpCommon                    0.2.6
 - HttpParser                    0.2.0
 - HttpServer                    0.1.6
 - Interpolations                0.3.6
 - Iterators                     0.1.10
 - JSON                          0.7.1
 - JuliaParser                   0.7.4
 - Juno                          0.2.2
 - KernelDensity                 0.3.0
 - LNR                           0.0.2
 - LaTeXStrings                  0.2.0
 - Lazy                          0.11.2
 - LegacyStrings                 0.1.1
 - Loess                         0.0.7
 - MachO                         0.0.4
 - MacroTools                    0.3.2
 - MbedTLS                       0.3.0
 - Measures                      0.0.3
 - Media                         0.2.2
 - Mustache                      0.1.2
 - Mux                           0.2.1
 - NaNMath                       0.2.1
 - NullableArrays                0.0.9
 - ObjFileBase                   0.0.4
 - Optim                         0.6.1
 - PDMats                        0.4.2
 - PositiveFactorizations        0.0.2
 - PyCall                        1.7.2
 - Ratios                        0.0.4
 - Reactive                      0.3.4
 - Reexport                      0.0.3
 - Requires                      0.3.0
 - Rmath                         0.1.3
 - SHA                           0.2.1
 - Showoff                       0.0.7
 - SortingAlgorithms             0.1.0
 - StatsBase                     0.10.0
 - StatsFuns                     0.3.1
 - StructIO                      0.0.2
 - TerminalUI                    0.0.2
 - TextWrap                      0.1.6
 - URIParser                     0.1.6
 - VT100                         0.0.2
 - WeakRefStrings                0.2.0
 - WebSockets                    0.2.1
 - WoodburyMatrices              0.2.0
```

**Comment by quinnj** [comment]
Trying to figure out what's going on here...

**Comment by quinnj** [comment]
So I've tracked this down to a seemingly non-Feather.jl problem; I'm guessing with regards to the # of bytes we're trying to write at once.
```julia
julia> a = open("testfile", "w")
IOStream(<file testfile>)

julia> A = zeros(UInt8, 118685760);

julia> position(a)
0

julia> write(a, A)
118685760

julia> position(a)
0

julia> flush(a)

julia> position(a)
0
```

**Comment by quinnj** [comment]
Found a work-around for now. Fixed in 62f5094fbd8e3f987c212e772249ecd958751bce
